nyx_client/nyx_client/circles.py

```diff
@@ -0,0 +1,113 @@
+# Copyright © 2024 IOTIC LABS LTD. info@iotics.com
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://github.com/Iotic-Labs/nyx-sdk/blob/main/LICENSE
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class RemoteHost:
+    """Represents a remote host on the network."""
+
+    def __init__(self, name: str, did: str):
+        """Builds a RemoteHost.
+
+        Args:
+            name: the name of the organization
+            did: the did of the remote host
+        """
+        self.name = name
+        self.did = did
+
+    def __repr__(self):
+        return f"RemoteHost({self.name}, {self.did})"
+
+
+class Circle:
+    """Represents a circle, which is a grouping of remote hosts."""
+
+    def __init__(
+        self, did: str, name: str, description: str | None = None, organizations: list[RemoteHost] | None = None
+    ):
+        """Builds a circle.
+
+        Args:
+            did: the did of the circle
+            name: the name of the circle
+            description: optional description of what the circle is
+            organizations: optional list of remote organizations in the circle
+        """
+        self.did = did
+        self.name = name
+        self.description = description
+        self.organizations = organizations
+
+    def __repr__(self):
+        return f"Circle({self.name}, {self.description}, {self.organizations})"
+
+    def contains(self, to_check: RemoteHost | str) -> bool:
+        """Checks if provided host is in the circle.
+
+        Args:
+            to_check: Either the RemoteHost to check, or a DID, or org name.
+
+        Returns:
+            Boolean showing if the remote host is included.
+        """
+        if not self.organizations:
```

Comment:
(not really necessary - for loops below will finish immeidatley if empty as long as orgs field cannot be None)